### DISCUSSION TITLE
XML v JSON and getting started

### DISCUSSION THREAD

**Discussion by mermicode** [maxGraph/maxGraph#60] [Q&A]
Hey All,

I recently found this project and I've been reading the threads and going through everything, wow this is a really cool project and cool community!

I'm junior/mid kind of level of dev on React - not sure I could add as much value as some of you guys but I'd love to get involved :)

I have a couple of question that really wanted to ask, been trying to find out through the discussions, apologies if this has been asked before.
1. does maxGraph use XML to render the diagrams or JSON?
2. this is a newb question, but how can I run maxGraph? I've been trying to run it and it come up with Storybook. Is there a way to run it as is?

many thanks :)

**Comment by tbouffard** [comment]
Hello @mermicode and welcome
1st be aware that the migration from mxGraph is still in progress, in particular, the documentation has not been updated at all.
If you need information about general concepts, I suggest you to read the [mxGraph manual](https://jgraph.github.io/mxgraph/docs/manual.html). The class names, the lib initialization changed, but this can help.

**about question 1**
If you question is about how to pass data to maxGraph, you generally create the diagram with JS/TS. maxGraph stores the relevant information internally in its own model. The model can be imported/exported through a custom XML format. 
mxGraph example: https://github.com/jgraph/mxgraph/blob/v4.2.2/javascript/examples/codec.html#L42. I have seen the migrated version in https://github.com/maxGraph/maxGraph/tree/development/packages/html/stories. The codec code is here: https://github.com/maxGraph/maxGraph/blob/74b67a60a4bf1fb3ffe7710e52100c6c9668ffe9/packages/core/src/util/serialization/mxCodec.js

**about question 2**
It should be possible to load maxGraph directly in a web page or import it in npm project. I don't know yet if the build generates the corresponding artifacts. For npm, we didn't publish anything publicly, so you will have to build it by yourself for now.  
If you still have question, please create a dedicated discussion for a better followup
